File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/minio/MinioProperties.java
package com.jmsoftware.maf.springcloudstarter.minio;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

/**
 * <h1>MinioProperty</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/7/21 9:39 PM
 */
@Data
@Validated
@ConfigurationProperties(prefix = MinioProperties.PREFIX)
public class MinioProperties {
    /**
     * The constant PREFIX.
     */
    public static final String PREFIX = "minio";
    /**
     * The Enabled.
     */
    @NotNull
    private Boolean enabled = Boolean.FALSE;
    /**
     * The Endpoint.
     */
    @NotBlank
    private String endpoint;
    /**
     * The Port.
     */
    @NotNull
    private Integer port;
    /**
     * The Access key. User ID
     */
    @NotBlank
    private String accessKey;
    /**
     * The Secret key. Password
     */
    @NotBlank
    private String secretKey;
    /**
     * The Secure.
     */
    private Boolean secure = Boolean.FALSE;
    /**
     * The Bucket name.
     */
    private String bucketName;
    /**
     * The Config dir.
     */
    private String configDir;
}

File: spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/springcloudstarter/minio/MinioProperties.kt
package com.jmsoftware.maf.springcloudstarter.minio

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.validation.annotation.Validated

/**
 * # MinioProperty
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 3:43 PM
 */
@Validated
@Configuration
@ConfigurationProperties(prefix = MinioProperties.PREFIX)
class MinioProperties {
    companion object {
        /**
         * The constant PREFIX.
         */
        const val PREFIX = "minio"
    }

    /**
     * The Enabled.
     */
    var enabled = false

    /**
     * The Endpoint.
     */
    lateinit var endpoint: @NotBlank String

    /**
     * The Port.
     */
    var port: @NotNull Int = 0

    /**
     * The Access key. User ID
     */
    lateinit var accessKey: @NotBlank String

    /**
     * The Secret key. Password
     */
    lateinit var secretKey: @NotBlank String

    /**
     * The Secure.
     */
    var secure = false

    /**
     * The Bucket name.
     */
    var bucketName: String? = null

    /**
     * The Config dir.
     */
    var configDir: String? = null
}
